README.md

```diff
@@ -1,8 +1,106 @@
+
+Since futarchy markets are driven by future price predicion, we use the [Gnosis DutchX](https://blog.gnosis.pm/introducing-the-gnosis-dutch-exchange-53bd3d51f9b2) as an oracle to resolve the price of the "FCR Token".
+
+## Setup
+
+`npm install`
+
+`npm run compile` to compile truffle artifacts to `build/contracts`
+
+## Smart Contracts
+
+### Registry.sol
+
+Contains core registry functionality, and stores the state of registry listings. Adapted from [skmgoldin/tcr/contracts/Registry.sol](https://github.com/skmgoldin/tcr/blob/4853e4dd8d6080d95e51955c81c1052017b1d3ec/contracts/Registry.sol).
+
+### RegistryFactory.sol
+
+Factory contract for creation of Registry instances.
+
+### Parameterizer.sol
+
+From [skmgoldin/tcr/contracts/Parameterizer](https://github.com/skmgoldin/tcr/blob/4853e4dd8d6080d95e51955c81c1052017b1d3ec/contracts/Parameterizer.sol). Used to set parameters for registries and challenges, such as `minDeposit`. New parameters values can be proposed with a token weighted vote. The voting process is implemented with [PLCRVoting](https://github.com/ConsenSys/PLCRVoting/blob/2e63c1f5190a9a1166b4b72f86e325ed61460b24/contracts/PLCRVoting.sol).
+
+### ParameterizerFactory.sol
+
+Factory contract for creation of Parameterizer instances.
+
+### IDutchExchange.sol
+
+Interface for [DutchExchange](https://github.com/gnosis/dx-contracts/blob/9c56c471df86587473c12aa057b87e44c34dd702/contracts/DutchExchange.sol) contract. Provides the functions that are needed for the FCR implementation, so that dutch exchange can be mocked for testing.
+
+### Imports.sol
+
+Used to force compilation of all contracts for testing. Not intended to be deployed to testnet/mainnet.
+
+### Challenge/FutarchyChallenge.sol
+
+The futarchy based challenge implementation. For each new challenge issued via Registry, there will be one instance of FutarchyChallenge deployed.
+
+The `initiateFutarchy()` function is used to create an instance of [FutarchyOracle](https://github.com/levelkdev/pm-contracts/blob/78ba5acb928381d3335b6cff851d16f7d61e8afd/contracts/Oracles/FutarchyOracle.sol), which creates all events and markets for the futarchy based challenge decision.
+
+### Challenge/FutarchyChallengeFactory.sol
+
+Factory contract for creation of FutarchyChallenge instances.
+
+### Challenge/ChallengeInterface.sol
+
+Interface for a challenge instance. Implemented by FutarchyChallenge and used by FutarchyChallengeFactory.
+
+### Challenge/ChallengeFactoryInterface.sol
+
+Interface for a challenge factory instance. Implemented by FutarchyChallengeFactory and used by Registry.
+
+### Challenge/Oracles/CentralizedTimedOracle.sol
```

Comment:
`CentralizedTimedOracle` can proooobably be deleted? I don't see much use for it here anymore.

Reply:
@emaG3m cool .. let me remove in a follow up after the readme gets added. If there's other stuff you think we should remove, let me know!